frontend/packages/client/src/components/ProposalsList/ProposalCardFooter.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useMemo, useState } from "react";
+import millify from "millify";
+import { Active, CheckCircle } from "../Svg";
+import { parseDateFromServer } from "utils";
+import StatusLabel from "../StatusLabel";
+import { useVotingResults } from "hooks";
+import { FilterValues } from "const";
+
+const ProposalCardFooter = ({ id, voted, endTime, computedStatus, isDesktopOnly }) => {
+  const [justMounted, setJustMounted] = useState(true);
+
+  const { getVotingResults, data: votingResults } = useVotingResults();
+
+  const { diffDuration } = parseDateFromServer(endTime);
+
+  const status = FilterValues[computedStatus] ?? FilterValues.closed;
+
+  useEffect(() => {
+    if (justMounted && (FilterValues[computedStatus] === FilterValues.closed)) {
+      async function _getVotingResults() {
+        return getVotingResults(id);
+      }
+      _getVotingResults();
+    }
+  }, [computedStatus, getVotingResults, id, justMounted]);
+
+  // prevents memory leak in above useEffect for getVotingResults
+  useEffect(() => () => { setJustMounted(false) }, []);
+
+  const { textDecision, winCount } = useMemo(() => {
+    if (votingResults?.results) {
+      const { results } = votingResults;
+      const resultsArray = Object.entries(results);
+      const sortedResults = resultsArray.sort((a, b) => {
+        if (Number(b[1]) < Number(a[1])) {
+          return -1;
+        }
+        return 1;
+      });
+
+      const textDecision = sortedResults[0][0];
+      const winCount = Number(sortedResults[0][1]);
+
+      // no winner
+      if (sortedResults.every((e) => Number(e[1]) === winCount)) {
+        return { textDecision: "", winCount: 0 };
+      }
+
+      return {
+        textDecision,
+        winCount,
+      };
+    }
+    return { textDecision: "", winCount: 0 };
+  }, [votingResults]);
+
+  const iconStatusMap = {
+    [FilterValues.active]: <Active />,
+    [FilterValues.closed]:
+      textDecision !== "" ? <CheckCircle width="15" height="15" /> : null,
+  };
+
+  const textDescriptionMap = {
+    [FilterValues.active]: `${diffDuration} left`,
+    [FilterValues.closed]:
+      textDecision !== ""
+        ? `${textDecision} ${winCount !== 0 ? `(${millify(winCount)})` : ""}`
+        : "",
+  };
+
+  const statusLabelMap = {
+    [FilterValues.active]: <StatusLabel status="Active" voted={voted} rounder
```

Comment:
Note: We don't have any set line-length wrapping rules, but I wrapped attributes that would go over 100 chars, for readability, in all files. Can change this if preferred.